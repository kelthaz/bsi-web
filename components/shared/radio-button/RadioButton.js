import React from 'react';
import PropTypes from 'prop-types';
import styles from './radio-button.module.scss';

const RadioButton = ({ name, onChange, value, children, label }) => (
  <div className={`${!children ? styles['container-check-text-no-children'] : styles['container-check-text']}`}>
    <div className={`${styles['container-input']}`}>
      <input
        name={name}
        className={` ${styles['my-check']}`}
        type="radio"
        value={label}
        onChange={onChange}
        checked={value === label}
      />
    </div>
    {children && <div className={`${styles['container-content']}`}>{children}</div>}
  </div>
);

RadioButton.defaultProps = {
  children: false,
};

RadioButton.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.any,
};

export default RadioButton;
