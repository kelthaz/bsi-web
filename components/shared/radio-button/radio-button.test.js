import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import RadioButton from './RadioButton';

describe('Pruebas en el componente RadioButton', () => {
  const onChange = jest.fn();
  test('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const props = {
      name: 'meEjercenControlMoralComoMoral',
      label: 'si',
      onChange,
      value: 'si',
    };
    // act
    const wrapper = shallow(
      <RadioButton {...props}>
        <p>Sí</p>
      </RadioButton>
    );
    // assert

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de llamarse onChange', () => {
    // arrange
    const valorEsperado = { target: { checked: true } };
    const props = {
      name: 'meEjercenControlMoralComoMoral',
      label: 'si',
      onChange,
      value: 'no',
    };
    // act
    const wrapper = shallow(
      <RadioButton {...props}>
        <p>Sí</p>
      </RadioButton>
    );
    wrapper.find('input').simulate('change', valorEsperado);
    // assert
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(valorEsperado);
  });
});
