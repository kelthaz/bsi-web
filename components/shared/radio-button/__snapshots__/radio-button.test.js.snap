// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente RadioButton Debe de mostrarse correctamente el componente 1`] = `
<div
  className="container-check-text"
>
  <div
    className="container-input"
  >
    <input
      checked={false}
      className=" my-check"
      name="meEjercenControlMoralComoMoral"
      onChange={[MockFunction]}
      type="radio"
      value="si"
    />
  </div>
  <div
    className="container-content"
  >
    <p>
      SÃ­
    </p>
  </div>
</div>
`;
